src/app/modules/main/chat_section/chat_content/messages/view.nim

```diff
@@ -128,6 +128,12 @@ QtObject:
   proc setEditModeOn*(self: View, messageId: string) {.slot.} =
    self.model.setEditModeOn(messageId)
 
+  proc setEditModeOnLastMessage*(self: View, pubkey: string) {.slot.} =
+    let lastMessage = self.model.getLastItem()
+    if lastMessage != nil: 
+      if lastMessage.senderId == pubkey:
```

Comment:
> In Discord, pressing the up key gets our last message, even if it is a day old. That way, it always does something, unless you never wrote anything, of course.

Actuallly, it does not, which is why I have implemented it the same way.

But I'm happy to change this either way!

Reply:
I stand corrected. It does!
The reason it didn't do it for me was because there was none of my messages on screen (although I did have messages in the channel)

Alrighty, will update this.

Reply:
Maybe you have an old or a bad version of Discord, but here is my experience with it (you can't see the keyboard presses, but I press up arrow in the input at the bottom and escape in the input at the top at the end):

[Screencast 2022-12-21 10:45:14-up-arrow.webm](https://user-images.githubusercontent.com/11926403/208945292-866df91a-5a6b-46e5-b5b0-9279437b3364.webm)

Reply:
Oops, you added an other comment while I was commenting, now I look like someone who doesn't read lol

Reply:
Thank you for your effort @jrainville and sorry for the confusion

Reply:
what about scrolling?
On Discord - you scroll to your last message.
Here, during the test - edit is activated, but I didn't see the msg I'm editing, only after second up key click, but second up key click removes focus

Reply:
Yes, scrolling isn't considered here. This was really a just a quick fix for having **something**  as opposed to nothing.
Happy to look into scrolling as well.

The losing focus on second arrow-up keystroke is prolly because the same handling is triggered. I'll change it such that arrow-up is only honored when not editing a message

Reply:
@mprakhov I've updated the PR to scroll to the message that needs to be edited.
Tested this and it seems to work fine.

I've also looked into the thing that arrow-up causes focus drop on the item to be edited. I thought it was because the edit view uses the chat input as well. But it doesn't. I'm not sure what's going on with `StatusInput` that it would cause that.

So I'd propose we fix that in an additional PR by someone who's better understanding of that component (and specifically QML)